simd/src/Kokkos_SIMD.hpp

```diff
@@ -29,6 +29,10 @@
 #include <Kokkos_SIMD_AVX512.hpp>
 #endif
 
+#ifdef __ARM_NEON
```

Comment:
Why did you decide to guard this way rather than follow what we had done for AVX2 and AVX512?
(Not necessarily asking you to changer, just trying to understand if you considered it.  To be fair it seems that these arch macros are only used in the simd package at the moment if we exclude the legacy atomics that we are about to remove anyway.)

Reply:
This is a good question. I think Kokkos had pre-existing macros for AVX and I was able to determine which Intel microarchitectures had support for which extensions. It was less clear to me which ARM architectures support NEON, so it was convenient to let the compiler indicate this. I agree that consistency would be better, perhaps we should use vendor-specific macros for AVX detection as well and reduce the responsibility of Kokkos macros.

Reply:
We discussed briefly with Christian. We think what you did is in the right direction, that we will probably get rid of `KOKKOS_ARCH_AVX*` macros right after we drop the legacy atomics.  One advantage is that it will enable the appropriate SIMD support when users configure with `-DKokkos_ARCH_NATIVE=ON`

Reply:
Sounds great!

Reply:
I think I've mentioned this elsewhere, but I think it might be important to maintain a distinction between (in particular, but maybe more generally) building for a microarchitecture that supports AVX-512 and specifically using AVX-512 SIMD.

https://travisdowns.github.io/blog/2020/01/17/avxfreq1.html and some of his other posts give details on why a user may want to compile 'native', but still use AVX2 SIMD.

I wouldn't be surprised if at least one implementation of ARM SVE ends up presenting similar issues down the line.

Reply:
I get your point, but I believe supporting this falls outside the scope of this PR review.

Reply:
That was more directed at Damien's remark about selecting SIMD instruction set based on `-DKokkos_ARCH_NATIVE` and dropping separate configuration variables/macros.